GeeksCoreLibrary/Components/Account/Services/AccountsService.cs

```diff
@@ -447,5 +454,199 @@ public async Task<List<RoleModel>> GetUserRolesAsync(ulong userId, bool includeP
+            var amountOfDaysToRememberCookie = GetAmountOfDaysToRememberCookie(settings);
+            var cookieValue = await GenerateNewCookieTokenAsync(userId, mainUserId, !amountOfDaysToRememberCookie.HasValue || amountOfDaysToRememberCookie.Value <= 0 ? 0 : amountOfDaysToRememberCookie.Value, settings.EntityType, settings.SubAccountEntityType, role);
+            await SaveGoogleClientIdAsync(userId, settings);
+
+            var offset = (amountOfDaysToRememberCookie ?? 0) <= 0 ? (DateTimeOffset?)null : DateTimeOffset.Now.AddDays(amountOfDaysToRememberCookie.Value);
+            var currentContext = httpContextAccessor.HttpContext;
+            HttpContextHelpers.WriteCookie(currentContext, settings.CookieName, cookieValue, offset, isEssential: true);
+
+            await SaveLoginAttemptAsync(true, userId, extraDataForReplacements, settings);
+        }
+
+        /// <summary>
+        /// Gets the Google Client ID from the Google Analytics cookie and saved it.
+        /// </summary>
+        /// <param name="userIdForGoogleCid">The ID of the user to save the CID for.</param>
+        /// <param name="settings"></param>
+        public async Task SaveGoogleClientIdAsync(ulong userIdForGoogleCid, AccountCmsSettingsModel settings)
+        {
+            var request = httpContextAccessor.HttpContext?.Request;
+            var googleClientIdCookieValue = request?.Cookies[Constants.GoogleAnalyticsCookieName];
+            if (String.IsNullOrWhiteSpace(googleClientIdCookieValue))
+            {
+                return;
+            }
+
+            // GA1.2.1248174149.1587127355
+            var clientIdSplit = googleClientIdCookieValue.Split(new[] { '.' }, StringSplitOptions.RemoveEmptyEntries);
+
+            if (clientIdSplit.Length != 4)
+            {
+                var sanitizedGoogleClientIdCookieValue = googleClientIdCookieValue.Replace(Environment.NewLine, "").Replace("\n", "").Replace("\r", "");
+                logger.LogTrace($"Google Analytics cookie found ({Constants.GoogleAnalyticsCookieName}: {sanitizedGoogleClientIdCookieValue}), but is not a valid format. Expected to have 3 dots, but it has {clientIdSplit.Length - 1}", true);
+                return;
+            }
+
+            if (userIdForGoogleCid <= 0)
+            {
+                var sanitizedGoogleClientIdCookieValue = googleClientIdCookieValue.Replace(Environment.NewLine, "").Replace("\n", "").Replace("\r", "");
+                logger.LogTrace($"Google Analytics cookie found ({Constants.GoogleAnalyticsCookieName}: {sanitizedGoogleClientIdCookieValue}), but the user is not logged in and we did not create a new account (successfully).", true);
+                return;
+            }
+
+            var googleClientId = String.Join(".", clientIdSplit.Skip(2));
+            
+            
+            var wiserItemsService = ActivatorUtilities.CreateInstance<WiserItemsService>(serviceProvider, databaseConnection);
```

Comment:
Can you add a comment above this line to explain why the Wiser Items Service is retrieved this way? So about the database connection needing to be the same scope.